gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/strategies/MapStrategy.java

```diff
@@ -0,0 +1,170 @@
+    // read length
+    if ((typeByte & PropertyValue.FLAG_LARGE) == PropertyValue.FLAG_LARGE) {
+      length = inputView.readInt();
+    } else {
+      length = inputView.readShort();
+    }
+
+    byte[] rawBytes = new byte[length];
+
+    for (int i = 0; i < rawBytes.length; i++) {
+      rawBytes[i] = inputView.readByte();
+    }
+
+    PropertyValue key;
+    PropertyValue value;
+    Map<PropertyValue, PropertyValue> map = new HashMap<>();
+    ByteArrayInputStream byteStream = new ByteArrayInputStream(rawBytes);
+    DataInputStream inputStream = new DataInputStream(byteStream);
+    DataInputView internalInputView = new DataInputViewStreamWrapper(inputStream);
+
+    try {
+      while (inputStream.available() > 0) {
+        key = new PropertyValue();
+        key.read(internalInputView);
+
+        value = new PropertyValue();
+        value.read(internalInputView);
+
+        map.put(key, value);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error reading PropertyValue");
+    }
+
+    return map;
+  }
+
+  @Override
+  public int compare(Map value, Object other) {
+    throw new UnsupportedOperationException(
+      "Method compareTo() is not supported for Map;"
+    );
+  }
+
+  @Override
+  public boolean is(Object value) {
+    boolean keyIsPropertyValue = false;
+    boolean valIsPropertyValue = false;
+
+    if (value instanceof Map) {
+      // set to true so empty maps get handled correctly
+      keyIsPropertyValue = true;
+      valIsPropertyValue = true;
+      for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) value).entrySet()) {
+        keyIsPropertyValue = entry.getKey() instanceof PropertyValue;
+        valIsPropertyValue = entry.getValue() instanceof PropertyValue;
+        if (!keyIsPropertyValue || !valIsPropertyValue) {
+          break;
+        }
+      }
+    }
+
+    return keyIsPropertyValue && valIsPropertyValue;
+  }
+
+  @Override
+  public Class<Map<PropertyValue, PropertyValue>> getType() {
+    return (Class) Map.class;
+  }
+
+  @Override
+  public Map<PropertyValue, PropertyValue> get(byte[] bytes) {
+    PropertyValue key;
+    PropertyValue value;
+
+    Map<PropertyValue, PropertyValue> map = new HashMap<PropertyValue, PropertyValue>();
+
+    ByteArrayInputStream byteStream = new ByteArrayInputStream(bytes);
+    DataInputStream inputStream = new DataInputStream(byteStream);
+    DataInputView inputView = new DataInputViewStreamWrapper(inputStream);
+
+    try {
+      if (inputStream.skipBytes(PropertyValue.OFFSET) != PropertyValue.OFFSET) {
+        throw new RuntimeException("Malformed entry in PropertyValue List");
+      }
+      while (inputStream.available() > 0) {
+        key = new PropertyValue();
+        key.read(inputView);
+
+        value = new PropertyValue();
+        value.read(inputView);
+
+        map.put(key, value);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error reading PropertyValue");
```

Comment:
You can add a cause here.
```suggestion
      throw new RuntimeException("Error reading PropertyValue", e);
```